### PR TITLE
Version 2 test bond

### PR DESCRIPTION
**Pull request by kushal256**
- Moves dapp lib and src code into test/dapp_test 
- Add basic bonding test case

### CODE REVIEW

**Comment by kushal256** [comment]
Ready for review

File: test/dapp_test/src/util/MockContract.sol [hunk 1]
```diff
@@ -0,0 +1,380 @@
+// SPDX-License-Identifier: UNLICENSED
+
+//Copied from https://github.com/gnosis/mock-contract/blob/b3db814ac41a3b632d8441cd1fbfbfce54e056f7/contracts/MockContract.sol but removed call "encodeWithSignature("updateInvocationCount" towards end
```
Comment (lolchocotaco) [inline test/dapp_test/src/util/MockContract.sol:3]: is there anyway we can include as dep? or is this better?

**Review by lolchocotaco — APPROVED** [review]
me thinks it looks good.

File: test/dapp_test/src/util/MockContract.sol (hunk 1 above)
Comment (kushal256) [reply test/dapp_test/src/util/MockContract.sol:3]: The reason why this was done is I had to do a modification, there was a bug in the usage with https://github.com/OlympusDAO/olympus-contracts/blob/Version-2/test/dapp_test/src/BondDepository.t.sol#L227

Perhaps I should add that comment here, too.

Looking at the github history of MockContract.sol, they had a switch to enable/disable this behavior, but then removed it so it always occurs.  I need to explore why they did that, but the project hasn't been touched in ages.  So for now thought this was the simplest path.  The downside is no way to do Verification (i.e. verify if a mock'd method was called).

**Pull request merged**